lib/charms/tls_certificates_interface/v2/tls_certificates.py

```diff
@@ -685,6 +685,7 @@ def generate_certificate(
     ca_key_password: Optional[bytes] = None,
     validity: int = 365,
     alt_names: Optional[List[str]] = None,
+    additional_extensions: Optional[List[x509.Extension]] = None,
```

Comment:
should we consider extracting some logic into new functions? this one has grown quite a bit.
For example would it make sense to handle the filtering of "default extensions" in a separate function?

Reply:
Removed `additional_extensions` arg.